File: convert_to_uppercase.py
import time

t1 = time.time()
with open("200mb (copy).txt", "r") as f:  # change file name for different files
    text = f.read().upper()
with open("200mb (copy).txt", "w") as f:  # name must be same as above file
    f.write(text)
t2 = time.time()
print(t2 - t1)

File: ConvertToUpperCase.java
import java.io.*;
import java.nio.file.*;
import java.nio.charset.StandardCharsets;

public class ConvertToUpperCase {
    public static void main(String[] args) {
        long t1 = System.currentTimeMillis();

        try {
            // Read the file
            Path path = Paths.get("1000mb (copy).txt");
            byte[] bytes = Files.readAllBytes(path);
            String text = new String(bytes, StandardCharsets.UTF_8);

            // Convert to uppercase
            text = text.toUpperCase();

            // Write the file
            Files.write(path, text.getBytes(StandardCharsets.UTF_8));
        } catch (IOException e) {
            e.printStackTrace();
        }

        long t2 = System.currentTimeMillis();
        System.out.println((t2 - t1) / 1000.0 + " seconds");
    }
}
